keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/DependsOnConstraintEvaluator.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.keel.constraints
+
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.persistence.PromotionRepository
+import org.springframework.stereotype.Component
+
+@Component
+class DependsOnConstraintEvaluator(
+  private val promotionRepository: PromotionRepository
+) : ConstraintEvaluator<DependsOnConstraint> {
+
+  override val constraintType = DependsOnConstraint::class.java
+
+  override fun canPromote(
+    artifact: DeliveryArtifact,
+    version: String,
+    deliveryConfig: DeliveryConfig,
+    targetEnvironment: String
+  ): Boolean {
+    val target = deliveryConfig.environments.firstOrNull { it.name == targetEnvironment }
+    requireNotNull(target) {
+      "No environment named $targetEnvironment exists in the configuration ${deliveryConfig.name}"
+    }
+
+    val constraint = target
+      .constraints
+      .filterIsInstance<DependsOnConstraint>()
+      .firstOrNull()
+
+    return if (constraint == null) {
+      true
```

Comment:
I would like to genericize this behavior. If the environment doesn't _have_ my supported constraint type, I should just approve any version.

Reply:
Can we do this where each constraint has a "handles" function which is the type of constraint it has an opinion on, and it only gets asked for an opinion if the env has a constraint of its type?